Source/WebInspectorUI/UserInterface/Models/LocalResourceOverride.js

```
@@ -346,7 +346,11 @@ WI.LocalResourceOverride = class LocalResourceOverride extends WI.Object
     get _urlRegex()
     {
         console.assert(this._isRegex);
-        return new RegExp(this._url, !this._isCaseSensitive ? "i" : "");
+        try {
+            return new RegExp(this._url, !this._isCaseSensitive ? "i" : "");
```

this seems like the wrong place to check.  we shouldn't even have a `WI.LocalResourceOverride` with invalid information in the first place (i.e. would it just never match anything and effectively be silently ignored?)

we already do other validation inside `WI.LocalResourceOverridePopover.prototype.get serializedData` so i think we should have something like this in there
```js
if (data.isRegex) {
    try {
        new RegExp(data.url);
    } catch {
        return null;
    }
}
```

---

>(i.e. would it just never match anything and effectively be silently ignored?)
Yes, that's what happens.

The user experience isn't great as-is, but it allows a developer to correct the mistake.

If we return `null` for `WI.LocalResourceOverridePopover.prototype.serializedData`, the Local Override won't even be created in `WI.LocalResourceOverrideTreeElement.willDismissPopover()` without any warning why.

One extra reason for keeping the fix here is to recover Web Inspector out of a bad state for people who ended up committing the bad RegEx to storage, like the [original reporter](https://developer.apple.com/forums/thread/772801).

---

If anything, I'd keep the solution in both places. 

Having an unguarded `new RegExp()` in `WI.LocalResourceOverride` is bound to bite again in the future if we offer another way to create it instead of free-form user input.

---

the typical pattern in Web Inspector is that by the time we're in the `constructor` (and ideally even earlier than that like in the "view" layer) all of the arguments should already have been checked to be valid

if the developer has something in their storage that's invalid, id suggest we check it there

as such, i think we instead should do something like this (in addition to the change you already made in `WI.LocalResourceOverridePopover.prototype.get serializedData`)
```js
let localResourceOverride = WI.LocalResourceOverride.fromJSON(serializedLocalResourceOverride);

if (localResourceOverride.isRegex) {
    // Temporary fix for if the local override was created before invalid URL regex checking was added.
    try {
        localResourceOverride._urlRegex;
    } catch {
        const key = null;
        WI.objectStores.localResourceOverrides.associateObject(localResourceOverride, key, serializedLocalResourceOverride);
        WI.objectStores.localResourceOverrides.deleteObject(localResourceOverride);
    }
}
```

---

This makes sense and helps to restore Web Inspector for users who accidentally ended up in this state. Updated PR 👍